icu4c/source/test/intltest/ustrtest.cpp

```diff
@@ -2310,3 +2311,21 @@ void UnicodeStringTest::TestUnicodeStringInsertAppendToSelf() {
     str.insert(2, sub);
     assertEquals("", u"abbcdcde", str);
 }
+
+void UnicodeStringTest::TestLargeAppend() {
+    IcuTestErrorCode status(*this, "TestLargeAppend");
+    // Make a large UnicodeString
+    int32_t len = 0xAFFFFFF;
+    UnicodeString str;
+    char16_t *buf = str.getBuffer(len);
+    for (int32_t i = 0; i < len; i++) {
```

Comment:
This test looks very slow: a loop for 0xAFFFFFF iterations.  Please either restrict this test to exhaustive mode or make it faster so that it can run in under 10ms.

Reply:
done